src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeCookingTime.java

```diff
@@ -0,0 +1,104 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.CookingRecipe;
+import org.bukkit.inventory.Recipe;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe;
+import org.skriptlang.skript.bukkit.recipes.MutableRecipe.MutableCookingRecipe;
+import org.skriptlang.skript.bukkit.recipes.RegisterRecipeEvent;
+
+@Name("Recipe Cooking Time")
+@Description("The cooking time of a blasting, furnace, campfire or smoking recipe.")
+@Examples({
+	"set {_recipe} to a new blasting recipe with the key \"my_recipe\":",
+		"\tset the recipe input item to raw gold named \"Impure Gold\"",
+		"\tset the recipe cooking time to 10 seconds",
+		"\tset the recipe result to gold ingot named \"Pure Gold\""
+})
+@Since("INSERT VERSION")
+public class ExprRecipeCookingTime extends PropertyExpression<Recipe, Timespan> {
+
+	static {
+		Skript.registerExpression(ExprRecipeCookingTime.class, Timespan.class, ExpressionType.PROPERTY,
+			"[the] recipe cook[ing] time [of %recipes%]",
+			"%recipes%'[s] recipe cook[ing] time");
+	}
+
+	private boolean isEvent = false;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[0].isDefault()) {
+			if (exprs[0] == null) {
+				Skript.error("There is no recipe in a '" + getParser().getCurrentEventName() + "' event.");
+				return false;
+			}
```

Comment:
this is already handled by the parser; i mentioned this in the last review
applies to all expressions with this
also it's very useless considering you just called isDefault on exprs[0], so an NPE would already be thrown.